content_handler/BUILD.gn

```diff
@@ -36,73 +31,54 @@ template("flutter_content_handler") {
       "runtime_holder.h",
       "service_protocol_hooks.cc",
       "service_protocol_hooks.h",
-      "software_rasterizer.cc",
-      "software_rasterizer.h",
+      "session_connection.cc",
+      "session_connection.h",
+      "vulkan_rasterizer.cc",
+      "vulkan_rasterizer.h",
+      "vulkan_surface_producer.cc",
+      "vulkan_surface_producer.h",
     ]
 
     deps = [
-      "//application/lib/app",
-      "//application/lib/svc",
-      "//apps/icu_data/lib",
-      "//apps/mozart/lib/flutter/sdk_ext",
-      "//apps/mozart/lib/skia:vmo",
-      "//apps/mozart/services/buffers",
-      "//apps/mozart/services/buffers/cpp",
-      "//apps/mozart/services/composition",
-      "//apps/mozart/services/input",
-      "//apps/mozart/services/views",
-      "//apps/tracing/lib/trace:provider",
-
-      # TODO(abarth): We shouldn't need to depend on libdart_builtin but we fail
-      # to link otherwise.
-      "//dart/runtime/bin:libdart_builtin",
-      "//dart/runtime/vm:libdart_platform",
-      "//flutter/assets",
-      "//flutter/common",
-      "//flutter/flow",
-      "//flutter/glue",
-      "//flutter/lib/ui",
-      "//flutter/runtime",
-      "//flutter/sky/engine/platform",
-      "//lib/fidl/dart/sdk_ext",
-      "//lib/ftl",
-      "//lib/mtl",
-      "//lib/tonic/mx",
-      "//lib/zip",
-      "//third_party/rapidjson",
-      "//third_party/skia",
-    ]
-    deps += extra_deps
-
-    if (flutter_enable_vulkan) {
-      defines += [ "FLUTTER_ENABLE_VULKAN=1" ]
-
-      if (flutter_use_vulkan_native_surface) {
-        defines += [ "FLUTTER_USE_VULKAN_NATIVE_SURFACE=1" ]
-        sources += [
-          "vulkan_native_rasterizer.cc",
-          "vulkan_native_rasterizer.h",
-        ]
-        libs += [ "hid" ]
-      } else {
-        sources += [
-          "vulkan_rasterizer.cc",
-          "vulkan_rasterizer.h",
-        ]
-      }
-
-      deps += [
-        "//flutter/vulkan",
-        "//magma:vulkan",
-      ]
-    }
+             "//application/lib/app",
+             "//application/lib/svc",
+             "//apps/icu_data/lib",
+             "//apps/mozart/lib/flutter/sdk_ext",
+             "//apps/mozart/lib/scene:client",
+             "//apps/mozart/lib/skia:vmo",
```

Comment:
I think you should remove skia::vmo unless we're using SkData from VMO or something, probably don't need it.